apigw-sns-filter-policy-lambda-sls-py/functions/process_bar_status.py

```diff
@@ -0,0 +1,16 @@
+import json
+
+from aws_lambda_powertools import Logger
+from aws_lambda_powertools.utilities.data_classes import event_source, SNSEvent
+from aws_lambda_powertools.utilities.typing import LambdaContext
+
+logger = Logger()
+
+
+@event_source(data_class=SNSEvent)
+@logger.inject_lambda_context(log_event=True)
+def lambda_handler(event: SNSEvent, context: LambdaContext):
+    for record in event.records:
+        message = record.sns.message
+        message_json = json.loads(message)
+        logger.info(f"Notification received for Foo message: {message_json}")
```

Comment:
```suggestion
        logger.info(f"Notification received for Bar message: {message_json}")
```

Reply:
I think it might have been easier to just use one function and set the type (foo/bar) via environment variable instead of duplicating the code. What do you think?